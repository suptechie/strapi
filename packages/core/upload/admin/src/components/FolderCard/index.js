export { FolderCard } from './FolderCard';
export { FolderCardBody } from './FolderCardBody';
export { FolderCardCheckbox } from './FolderCardCheckbox';
export { FolderCardLink } from './FolderCardLink';
