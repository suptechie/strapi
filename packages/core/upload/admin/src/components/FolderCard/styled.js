import styled from 'styled-components';

import { BaseLink } from '@strapi/design-system/BaseLink';

export const FolderCardLink = styled(BaseLink)`
  text-decoration: none;
`;
