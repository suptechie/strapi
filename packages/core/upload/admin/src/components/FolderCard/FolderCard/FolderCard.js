import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { NavLink } from 'react-router-dom';

import { pxToRem } from '@strapi/helper-plugin';
import { Box } from '@strapi/design-system/Box';
import { CardAction } from '@strapi/design-system/Card';
import { Stack } from '@strapi/design-system/Stack';
import Folder from '@strapi/icons/Folder';

import { FolderCardContext } from '../contexts/FolderCard';
import useId from '../hooks/useId';

const FauxClickWrapper = styled.button`
  height: 100%;
  left: 0;
  position: absolute;
  opacity: 0;
  top: 0;
  width: 100%;

  &:hover,
  &:focus {
    text-decoration: none;
  }
`;

const StyledFolder = styled(Folder)`
  path {
    fill: currentColor;
  }
`;

const CardActionDisplay = styled(Box)`
  display: none;
`;

const Card = styled(Box)`
  &:hover,
  &:focus-within {
    ${CardActionDisplay} {
      display: ${({ isCardActions }) => (isCardActions ? 'block' : '')};
    }
  }
`;

export const FolderCard = ({
  children,
  id,
  startAction,
  cardActions,
  ariaLabel,
  onClick,
  to,
  ...props
}) => {
  const generatedId = useId(id);

  return (
    <FolderCardContext.Provider value={{ id: generatedId }}>
      <Card position="relative" tabIndex={0} isCardActions={!!cardActions} {...props}>
        <FauxClickWrapper
          to={to || undefined}
          as={to ? NavLink : 'button'}
          type={to ? undefined : 'button'}
          onClick={onClick}
          tabIndex={-1}
          aria-label={ariaLabel}
          aria-hidden
        />

        <Stack
          hasRadius
          background="neutral0"
          shadow="tableShadow"
          paddingBottom={3}
          paddingLeft={4}
          paddingRight={4}
          paddingTop={3}
          spacing={3}
          horizontal
          cursor="pointer"
        >
          {startAction}

          <Box
            hasRadius
            background="secondary100"
            color="secondary500"
            paddingBottom={2}
            paddingLeft={3}
            paddingRight={3}
            paddingTop={2}
          >
            <StyledFolder width={pxToRem(20)} height={pxToRem(18)} />
          </Box>

          {children}

          <CardActionDisplay>
            <CardAction right={4}>{cardActions}</CardAction>
          </CardActionDisplay>
        </Stack>
      </Card>
    </FolderCardContext.Provider>
  );
};

FolderCard.defaultProps = {
  id: undefined,
  cardActions: null,
  startAction: null,
  to: undefined,
};

FolderCard.propTypes = {
  ariaLabel: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  id: PropTypes.string,
  onClick: PropTypes.func.isRequired,
  startAction: PropTypes.element,
  cardActions: PropTypes.element,
  to: PropTypes.string,
};
