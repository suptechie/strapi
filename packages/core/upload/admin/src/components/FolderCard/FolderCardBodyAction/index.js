import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import styled from 'styled-components';

const FolderCardLink = styled(NavLink)`
  max-width: 100%;
  text-decoration: none;
`;

export const FolderCardBodyAction = ({ to, ...props }) => {
  if (to) {
    return <FolderCardLink to={to} {...props} />;
  }

  return <button type="button" {...props} />;
};

FolderCardBodyAction.defaultProps = {
  to: undefined,
};

FolderCardBodyAction.propTypes = {
  to: PropTypes.string,
};
