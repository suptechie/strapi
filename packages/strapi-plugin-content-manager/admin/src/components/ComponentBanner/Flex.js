import styled from 'styled-components';

const Flex = styled.div`
  display: flex;
  > button {
    cursor: pointer;
    padding-top: 0;
  }
  .trash-icon {
    color: #4b515a;
  }
  .button-wrapper {
    line-height: 35px;
  }
`;

export default Flex;
