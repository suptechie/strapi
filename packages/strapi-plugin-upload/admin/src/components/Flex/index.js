import styled from 'styled-components';
import PropTypes from 'prop-types';

const Flex = styled.div`
  display: flex;
  justify-content: ${({ justifyContent }) => justifyContent};
  flex-direction: ${({ flexDirection }) => flexDirection};
`;

Flex.defaultProps = {
  justifyContent: 'normal',
  flexDirection: 'row',
};

Flex.propTypes = {
  justifyContent: PropTypes.string,
  flexDirection: PropTypes.string,
};

export default Flex;
