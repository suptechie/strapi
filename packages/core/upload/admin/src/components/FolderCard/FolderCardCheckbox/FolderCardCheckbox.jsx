import React from 'react';

import { Checkbox, Box } from '@strapi/design-system';

import { useFolderCard } from '../contexts/FolderCard';

export const FolderCardCheckbox = (props) => {
  const { id } = useFolderCard();

  return (
    <Box position="relative" zIndex={2}>
      <Checkbox aria-labelledby={`${id}-title`} {...props} />
    </Box>
  );
};
