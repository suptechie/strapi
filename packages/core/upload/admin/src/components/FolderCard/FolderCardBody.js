import React from 'react';
import styled from 'styled-components';

import { Flex } from '@strapi/design-system/Flex';

import { useFolderCard } from './FolderCardContext';

const StyledBox = styled(Flex)`
  user-select: none;
`;

export const FolderCardBody = props => {
  const { id } = useFolderCard();

  return (
    <StyledBox
      {...props}
      id={`${id}-title`}
      alignItems="flex-start"
      direction="column"
      position="relative"
      zIndex={3}
    />
  );
};
