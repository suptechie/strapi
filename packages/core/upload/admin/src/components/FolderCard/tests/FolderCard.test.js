import React from 'react';
import { BaseLink } from '@strapi/design-system/BaseLink';
import { Flex } from '@strapi/design-system/Flex';
import { ThemeProvider, lightTheme } from '@strapi/design-system';
import { Typography } from '@strapi/design-system/Typography';
import { render, fireEvent, act } from '@testing-library/react';

import { FolderCard } from '../FolderCard';
import { FolderCardBody } from '../FolderCardBody';
import { FolderCardCheckbox } from '../FolderCardCheckbox';

const ID_FIXTURE = 'folder';

// eslint-disable-next-line react/prop-types
const ComponentFixture = props => {
  return (
    <ThemeProvider theme={lightTheme}>
      <FolderCard
        id={ID_FIXTURE}
        ariaLabel="Folder 1"
        startAction={<></>}
        onDoubleClick={() => {}}
        {...props}
      >
        <FolderCardBody as="h2">
          <BaseLink href="https://strapi.io" textDecoration="none">
            <Flex direction="column" alignItems="flex-start">
              <Typography variant="omega" fontWeight="semiBold">
                Pictures
              </Typography>
            </Flex>
          </BaseLink>
        </FolderCardBody>
      </FolderCard>
    </ThemeProvider>
  );
};

const setup = props => render(<ComponentFixture {...props} />);

describe('FolderCard', () => {
  test('renders and matches the snapshot', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  test('properly calls the onDoubleClick callback', () => {
    const callback = jest.fn();
    const { container } = setup({ onDoubleClick: callback });

    act(() => {
      fireEvent(container.querySelector('button'), new MouseEvent('dblclick', { bubbles: true }));
    });

    expect(callback).toHaveBeenCalledTimes(1);
  });

  test('has all required ids set when rendering a start action', () => {
    const { container } = setup({
      startAction: <FolderCardCheckbox value={false} />,
    });

    expect(container).toMatchSnapshot();
    expect(container.querySelector(`[id="${ID_FIXTURE}-3-title"]`)).toBeInTheDocument();
    expect(
      container.querySelector(`[aria-labelledby="${ID_FIXTURE}-3-title"]`)
    ).toBeInTheDocument();
  });
});
