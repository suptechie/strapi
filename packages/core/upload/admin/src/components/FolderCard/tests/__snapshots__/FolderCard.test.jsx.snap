// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FolderCard has all required ids set when rendering a start action 1`] = `
.c0 {
  position: relative;
}

.c2 {
  background: #ffffff;
  padding-block: 12px;
  padding-inline: 12px;
  border-radius: 4px;
  border-style: solid;
  border-width: 1px;
  border-color: #eaeaef;
  box-shadow: 0px 1px 4px rgba(33, 33, 52, 0.1);
  cursor: pointer;
}

.c4 {
  position: relative;
  z-index: 2;
}

.c6 {
  background: #eaf5ff;
  color: #66b7f1;
  padding-block-start: 8px;
  padding-inline-end: 12px;
  padding-block-end: 8px;
  padding-inline-start: 12px;
  border-radius: 4px;
}

.c8 {
  position: relative;
  overflow: hidden;
  max-width: 100%;
}

.c11 {
  padding-block: 4px;
  padding-inline: 4px;
  max-width: 100%;
}

.c15 {
  right: 16px;
}

.c13 {
  font-size: 1.4rem;
  line-height: 1.43;
  font-weight: 500;
  color: currentcolor;
}

.c3 {
  align-items: center;
  display: flex;
  flex-direction: row;
  gap: 8px;
}

.c9 {
  align-items: flex-start;
  display: flex;
  flex-direction: column;
}

.c17 {
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c5 {
  height: 18px;
  min-width: 18px;
  margin: 0;
  border-radius: 4px;
  border: 1px solid #c0c0cf;
  -webkit-appearance: none;
  background-color: #ffffff;
  cursor: pointer;
}

.c5:checked {
  background-color: #4945ff;
  border: 1px solid #4945ff;
}

.c5:checked:after {
  content: '';
  display: block;
  position: relative;
  background: url("data:image/svg+xml,%3csvg%20width='10'%20height='8'%20viewBox='0%200%2010%208'%20fill='none'%20xmlns='http://www.w3.org/2000/svg'%3e%3cpath%20d='M8.55323%200.396973C8.63135%200.316355%208.76051%200.315811%208.83931%200.395768L9.86256%201.43407C9.93893%201.51157%209.93935%201.6359%209.86349%201.7139L4.06401%207.67724C3.9859%207.75755%203.85707%207.75805%203.77834%207.67834L0.13866%203.99333C0.0617798%203.91549%200.0617102%203.79032%200.138504%203.7124L1.16213%202.67372C1.24038%202.59432%201.36843%202.59422%201.4468%202.67348L3.92174%205.17647L8.55323%200.396973Z'%20fill='white'%20/%3e%3c/svg%3e") no-repeat no-repeat center center;
  width: 10px;
  height: 10px;
  left: 50%;
  top: 50%;
  transform: translateX(-50%) translateY(-50%);
}

.c5:checked:disabled:after {
  background: url("data:image/svg+xml,%3csvg%20width='10'%20height='8'%20viewBox='0%200%2010%208'%20fill='none'%20xmlns='http://www.w3.org/2000/svg'%3e%3cpath%20d='M8.55323%200.396973C8.63135%200.316355%208.76051%200.315811%208.83931%200.395768L9.86256%201.43407C9.93893%201.51157%209.93935%201.6359%209.86349%201.7139L4.06401%207.67724C3.9859%207.75755%203.85707%207.75805%203.77834%207.67834L0.13866%203.99333C0.0617798%203.91549%200.0617102%203.79032%200.138504%203.7124L1.16213%202.67372C1.24038%202.59432%201.36843%202.59422%201.4468%202.67348L3.92174%205.17647L8.55323%200.396973Z'%20fill='%238E8EA9'%20/%3e%3c/svg%3e") no-repeat no-repeat center center;
}

.c5:disabled {
  background-color: #dcdce4;
  border: 1px solid #c0c0cf;
}

.c5:indeterminate {
  background-color: #4945ff;
  border: 1px solid #4945ff;
}

.c5:indeterminate:after {
  content: '';
  display: block;
  position: relative;
  color: white;
  height: 2px;
  width: 10px;
  background-color: #ffffff;
  left: 50%;
  top: 50%;
  transform: translateX(-50%) translateY(-50%);
}

.c5:indeterminate:disabled {
  background-color: #dcdce4;
  border: 1px solid #c0c0cf;
}

.c5:indeterminate:disabled:after {
  background-color: #8e8ea9;
}

.c16 {
  position: absolute;
  top: 12px;
}

.c1 {
  height: 100%;
  left: 0;
  position: absolute;
  opacity: 0;
  top: 0;
  width: 100%;
}

.c1:hover,
.c1:focus {
  text-decoration: none;
}

.c7 path {
  fill: currentColor;
}

.c14 {
  display: none;
}

.c10 {
  user-select: none;
}

.c12:focus {
  outline: 2px solid #4945ff;
  outline-offset: -2px;
}

<div>
  <div
    class="c0 "
    tabindex="0"
  >
    <button
      aria-hidden="true"
      aria-label="Folder 1"
      class="c1"
      tabindex="-1"
      tag="button"
      type="button"
    />
    <div
      class="c2 c3"
    >
      <div
        class="c4"
      >
        <div
          class=""
        >
          <input
            aria-labelledby="folder-3-title"
            class="c5"
            type="checkbox"
          />
        </div>
      </div>
      <div
        class="c6"
      >
        <svg
          class="c7"
          fill="currentColor"
          height="2.4rem"
          viewBox="0 0 32 32"
          width="2.4rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M27 9H16.414L13 5.586A1.98 1.98 0 0 0 11.586 5H5a2 2 0 0 0-2 2v18.078A1.926 1.926 0 0 0 4.924 27H27.11A1.89 1.89 0 0 0 29 25.111V11a2 2 0 0 0-2-2M5 7h6.586l2 2H5z"
          />
        </svg>
      </div>
      <h2
        class="c8 c9 c10"
        id="folder-3-title"
      >
        <button
          class="c11 c12"
          type="button"
        >
          <div
            class="c9"
          >
            <span
              class="c13"
            >
              Pictures
            </span>
          </div>
        </button>
      </h2>
      <div
        class="c14"
      >
        <div
          class="c15 c3 c16"
        />
      </div>
    </div>
  </div>
  <span
    class="c17"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </span>
</div>
`;

exports[`FolderCard renders and matches the snapshot 1`] = `
.c0 {
  position: relative;
}

.c2 {
  background: #ffffff;
  padding-block: 12px;
  padding-inline: 12px;
  border-radius: 4px;
  border-style: solid;
  border-width: 1px;
  border-color: #eaeaef;
  box-shadow: 0px 1px 4px rgba(33, 33, 52, 0.1);
  cursor: pointer;
}

.c4 {
  background: #eaf5ff;
  color: #66b7f1;
  padding-block-start: 8px;
  padding-inline-end: 12px;
  padding-block-end: 8px;
  padding-inline-start: 12px;
  border-radius: 4px;
}

.c6 {
  position: relative;
  overflow: hidden;
  max-width: 100%;
}

.c9 {
  padding-block: 4px;
  padding-inline: 4px;
  max-width: 100%;
}

.c13 {
  right: 16px;
}

.c11 {
  font-size: 1.4rem;
  line-height: 1.43;
  font-weight: 500;
  color: currentcolor;
}

.c3 {
  align-items: center;
  display: flex;
  flex-direction: row;
  gap: 8px;
}

.c7 {
  align-items: flex-start;
  display: flex;
  flex-direction: column;
}

.c15 {
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c14 {
  position: absolute;
  top: 12px;
}

.c1 {
  height: 100%;
  left: 0;
  position: absolute;
  opacity: 0;
  top: 0;
  width: 100%;
}

.c1:hover,
.c1:focus {
  text-decoration: none;
}

.c5 path {
  fill: currentColor;
}

.c12 {
  display: none;
}

.c8 {
  user-select: none;
}

.c10:focus {
  outline: 2px solid #4945ff;
  outline-offset: -2px;
}

<div>
  <div
    class="c0 "
    tabindex="0"
  >
    <button
      aria-hidden="true"
      aria-label="Folder 1"
      class="c1"
      tabindex="-1"
      tag="button"
      type="button"
    />
    <div
      class="c2 c3"
    >
      <div
        class="c4"
      >
        <svg
          class="c5"
          fill="currentColor"
          height="2.4rem"
          viewBox="0 0 32 32"
          width="2.4rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M27 9H16.414L13 5.586A1.98 1.98 0 0 0 11.586 5H5a2 2 0 0 0-2 2v18.078A1.926 1.926 0 0 0 4.924 27H27.11A1.89 1.89 0 0 0 29 25.111V11a2 2 0 0 0-2-2M5 7h6.586l2 2H5z"
          />
        </svg>
      </div>
      <h2
        class="c6 c7 c8"
        id="folder-1-title"
      >
        <button
          class="c9 c10"
          type="button"
        >
          <div
            class="c7"
          >
            <span
              class="c11"
            >
              Pictures
            </span>
          </div>
        </button>
      </h2>
      <div
        class="c12"
      >
        <div
          class="c13 c3 c14"
        />
      </div>
    </div>
  </div>
  <span
    class="c15"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </span>
</div>
`;
